app/src/main/java/com/eveningoutpost/dexdrip/SnoozeActivity.java

```diff
@@ -293,24 +299,57 @@ void displayStatus() {
+                status = "";
+            else
+                status = "No active alert exists";
             buttonSnooze.setVisibility(View.GONE);
             snoozeValue.setVisibility(View.GONE);
         } else {
             if(!aba.ready_to_alarm()) {
                 status = "Active alert exists named \"" + activeBgAlert.name + "\" Alert snoozed until " +
                     DateFormat.getTimeInstance(DateFormat.MEDIUM).format(new Date(aba.next_alert_at)) +
-                    " (" + (aba.next_alert_at - new Date().getTime()) / 60000 + " minutes left)";
+                    " (" + (aba.next_alert_at - now) / 60000 + " minutes left)";
             } else {
                 status = "Active alert exists named \"" + activeBgAlert.name + "\" (not snoozed)";
             }
             SetSnoozePickerValues(snoozeValue, activeBgAlert.above, activeBgAlert.default_snooze);
-            alertStatus.setText(status);
         }
 
+        //check if there are disabled alerts and if yes add warning
+        if (prefs.getLong("alerts_disabled_until", 0) > now) {
+            String textToAdd = (prefs.getLong("alerts_disabled_until", 0) > now + (infiniteSnoozeValueInMinutes - 365 * 24 * 60) * 60 * 1000)
+                    //if alerts would have been disabled "until you re-enable", and this test is done less than 365 * 24 * 60 minutes later, then this test will give true
+                    ? "you re-enable":DateFormat.getTimeInstance(DateFormat.MEDIUM).format(new Date(prefs.getLong("alerts_disabled_until", 0)));
+            status = "All alerts disabled until " + textToAdd;
+        } else {
+            if (prefs.getLong("low_alerts_disabled_until", 0) > now) {
+                String textToAdd = (prefs.getLong("low_alerts_disabled_until", 0) > now + (infiniteSnoozeValueInMinutes - 365 * 24 * 60) * 60 * 1000)
+                        //if low alerts would have been disabled "until you re-enable", and this test is done less than 365 * 24 * 60 minutes later, then this test will give true
+                        ? "you re-enable":DateFormat.getTimeInstance(DateFormat.MEDIUM).format(new Date(prefs.getLong("low_alerts_disabled_until", 0)));
+                status += "\n\nLow alerts disabled until " + textToAdd;
+            }
+            if (prefs.getLong("high_alerts_disabled_until", 0) > now) {
+                String textToAdd = (prefs.getLong("high_alerts_disabled_until", 0) > now + (infiniteSnoozeValueInMinutes - 365 * 24 * 60) * 60 * 1000)
+                        //if high alerts would have been disabled "until you re-enable", and this test is done less than 365 * 24 * 60 minutes later, then this test will give true
+                        ? "you re-enable":DateFormat.getTimeInstance(DateFormat.MEDIUM).format(new Date(prefs.getLong("high_alerts_disabled_until", 0)));
+                status += "\n\nHigh alerts disabled until " + textToAdd;
+            }
+        }
+        alertStatus.post(new Runnable() {
```

Comment:
Is there any reason to post that to the element.
We are on an UI thread anyways. This could lead to lags in updating the UI (seen with the Database-Import that also posts to the UI).